src/Types/Value.purs

```diff
@@ -0,0 +1,276 @@
+module Types.Value
+  ( CurrencySymbol(CurrencySymbol)
+  , TokenName(TokenName)
+  , Value(Value)
+  , allTokenNames
+  , emptyValue
+  , eq
+  , flattenValue
+  , geq
+  , getValue
+  , gt
+  , isAdaOnly
+  , isPos
+  , isZero
+  , leq
+  , lt
+  , minus
+  , numCurrencySymbols
+  , numCurrencySymbols'
+  , numTokenNames
+  , numTokenNames'
+  , singleton
+  , unflattenValue
+  , valueOf
+  )
+  where
+
+import Prelude
+import Control.Alternative (guard)
+import Data.Array (filter)
+import Data.BigInt (BigInt, fromInt)
+import Data.Foldable (any, length)
+import Data.Generic.Rep (class Generic)
+import Data.List ((:), all, foldMap, List(Nil))
+import Data.Map (keys, lookup, Map, toUnfoldable, unions, values)
+import Data.Map as Map
+import Data.Maybe (maybe, Maybe(Just, Nothing))
+import Data.Newtype (class Newtype, unwrap)
+import Data.Set (Set)
+import Data.Show.Generic (genericShow)
+import Data.These (These(Both, That, This))
+import Data.Tuple.Nested ((/\), type (/\))
+
+import Types.ByteArray (ByteArray)
+
+-- This module rewrites functionality from:
+-- https://github.com/mlabs-haskell/mlabs-pab/blob/master/src/MLabsPAB/PreBalance.hs
```

Comment:
Both links 404'd

Reply:
Ah, it's changed name, wonderful :D https://github.com/mlabs-haskell/bot-plutus-interface/blob/master/src/BotPlutusInterface/PreBalance.hs good spot.